File: 4252-first-unique-even-element/first-unique-even-element.cpp
class Solution {
public:
    int firstUniqueEven(vector<int>& nums) {

        unordered_map<int, int> mp;
        int n = nums.size();

        for(int i=0; i<n; i++) {
            mp[nums[i]]++;
        }

        for(int i=0; i<n; i++) {
            if (nums[i] % 2 == 0 && mp[nums[i]] == 1) return nums[i];
        }
        
        return -1;
    }
};

File: 4252-first-unique-even-element/first-unique-even-element.cs
public class Solution {
    public int FirstUniqueEven(int[] nums) {
        Dictionary<int, int> set = new();

        foreach(int x in nums){
            if(x%2 == 0){
                if(set.ContainsKey(x)){
                    set[x]+= 1;
                }
                else{
                    set[x] = 1;
                }
            }
        }

        foreach(int x in nums){
            if(x%2 ==0 && set[x] == 1){
                return x;
            }
        }

        return -1;
    }
}